CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -665,74 +772,249 @@ private void HandleBettingPhase()
                     HandlePreflop();
                     break;
                 case BettingPhase.FLOP:
-                    // HandleFlop();
+                    HandleFlop();
                     break;
                 case BettingPhase.TURN:
-                    // HandleTurn();
+                    HandleTurn();
                     break;
                 case BettingPhase.RIVER:
-                    // HandleRiver();
+                    HandleRiver();
                     break;
             }
         }
 
+        /// <summary>
+        /// handles the initial stage of betting before any community cards appear
+        /// </summary>
         private void HandlePreflop()
```

Comment:
I am very confused what is happening here and I think it will not be able to be integrated into the update loop.

Is the intention that all of this code happens at once in a single game frame? if so, it is fine BUT if it is intended that this is a broad method to handle multiple game frames or parts of the game, i.e., each player doing an action, this will NOT work because you are nesting a while loop within the main Update loop. Instead of this, you should organize this so that the ONLY loop is the Update loop.